### PR TITLE
feature gate: add to CI

### PR DESCRIPTION
**Pull request by buffalojoec**
This PR adds the Feature Gate program to the SPL repository's CI.

### CODE REVIEW

**Comment by buffalojoec** [comment]
Current dependencies on/for this PR:
* master
  * **PR #5605** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-program-library/5605" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-program-library/5605?utm_source=stack-comment).

File: .github/workflows/pull-request-feature-gate.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    paths:
+    - 'feature-gate/**'
+    - 'token/**'
+    - 'ci/*-version.sh'
+    - '.github/workflows/pull-request-feature-gate.yml'
+  push:
+    branches: [master]
+    paths:
+    - 'feature-gate/**'
+    - 'token/**'
```
Comment (joncinque) [inline .github/workflows/pull-request-feature-gate.yml:14]: Same here, no need to depend on token, right?

**Review by joncinque — COMMENTED** [review]
Just two little things, then this is good to go, thanks!

File: .github/workflows/pull-request-feature-gate.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+name: Feature Gate Pull Request
+
+on:
+  pull_request:
+    paths:
+    - 'feature-gate/**'
+    - 'token/**'
```
[the remark this answers is not in the document]
Comment (buffalojoec) [reply .github/workflows/pull-request-feature-gate.yml:7]: caught me copy-pasting!

**Review by joncinque — APPROVED** [review]

**Pull request merged**